src/wp-includes/link-template.php

```diff
@@ -4466,7 +4466,7 @@ function get_avatar_data( $id_or_email, $args = null ) {
 		$user = get_user_by( 'id', absint( $id_or_email ) );
 	} elseif ( is_string( $id_or_email ) ) {
 		if ( str_contains( $id_or_email, '@md5.gravatar.com' ) ) {
```

Comment:
Does this email need to change?

Reply:
> Does this email need to change?

This is in `src/wp-admin/includes/credits.php` to prevent double hashing, because `http://api.wordpress.org/core/credits/1.1/` API already returns a list of md5 hashes of the mails.

Maybe this API also needs to be modified to return a SHA256 hash?

Reply:
> Does this email need to change?

And we can use the third parameter of `get_avatar_data` to pass a flag to make email not need to hash, after that we can remove all the `@md5.gravatar.com`.